py3 port of bl-xdg-autostart

2ion opened the pull request:
Expands and replaces #107 .

 * Replace custom argument parsing with stdlib's argparse module
 * Remove meaningless --version switch
 * Remove meaningless guard against missing imports - we distribute the tool versioned + with dependencies, this is not needed

Apart from the interface change of removing --version, which I sincerely hope we don't use anywhere as the output is pretty useless, the tool's behaviour should match the original version.

There is an uncleanliness to the help text output if an invalid choice for the environment name is specified; this is because of https://github.com/BunsenLabs/bunsen-configs/pull/109/commits/ddfffa59a9022aaf0ddbaca05e61373db55ee8e0 which in turn is a workaround for the upstream bug https://bugs.python.org/issue9625 ; I think this small cosmetic issue is acceptable:

```
 $ python bin/bl-xdg-autostart --list ax
usage: bl-xdg-autostart [-h] [-l] [ENVIRONMENT [ENVIRONMENT ...]]
bl-xdg-autostart: error: argument ENVIRONMENT: invalid choice: 'ax' (choose from 'GNOME', 'KDE', 'ROX', 'XFCE', 'Old', [])
```
(note the last entry in "choose from").

jmechnich committed:
bl-xdg-autostart: basic port to python3

Commit:
bl-xdg-autostart: replace custom argument handling with ArgumentParser

Commit:
bl-xdg-autostart: remove --version because it has lost its meaning

Part of bunsen-configs, and versioned as a part thereof.

Commit:
bl-xdg-autostart: work around the infamous nargs with choices problem

See upstream bug https://bugs.python.org/issue9625.

johnraff commented on bin/bl-xdg-autostart:
```diff
@@ -21,27 +20,40 @@
+    )
+
+    ap.add_argument("-l", "--list",
+        help="""Show a list of the files which but be run. Files which would be run are marked
```
```suggestion
        help="""Show a list of the files which would be run. Files which would be run are marked
```

johnraff commented:
This looks mostly fine.

Agreed, considering this script could now fairly be considered a fork of the Openbox original, and none of our other short scripts have version numbers, then `--version' has no meaning.

See my suggestion above for one word replacement in `--help` output.

On BL Lithium:

`--list` works fine except for an unwanted line at the top of the output:
`(environment=[], list=True)`

Autostarting of applications at login time tested out OK.

Commit:
bl-xdg-autostart: remove printf debugging

ghost commented:
The extra line was a leftover from printf debugging, my mistake.

Fixed the help text.

Commit:
bl-xdg-autostart: fix typo in help text

johnraff commented on bin/bl-xdg-autostart:
```diff
@@ -21,27 +20,39 @@
+
+    ap.add_argument("-l", "--list",
+        help="""Show a list of the files which would be run. Files which would be run are marked
+with an asterix symbol [*]. For files which would not be run, information is given for
```
```suggestion
with an asterisk symbol [*]. For files which would not be run, information is given for
```

johnraff committed:
Update bin/bl-xdg-autostart

Trivial change in help text.

johnraff commented on bin/bl-xdg-autostart:
```diff
@@ -36,7 +36,7 @@ def main() -> int:
         help="""Show a list of the files which would be run. Files which would be run are marked
-with an asterix symbol [*]. For files which would not be run, information is given for
+with an asterisk symbol [*]. For files which would not be run, information is given for
 why they are excluded""",
```
```suggestion
why they are excluded.""",
```

johnraff committed:
Add full stop to help text.

johnraff commented:
Sorry, I missed another word and full stop in the help text - committed directly because they were trivial changes.

More important, I failed to notice some more unwanted content in the help output, perhaps coming from the upstream bug workaround. Both `positional arguments` and `optional arguments` have a meaningless and confusing (default: *) item:
```
john@lithium:~$ bl-xdg-autostart -h
usage: bl-xdg-autostart [-h] [-l] [ENVIRONMENT [ENVIRONMENT ...]]

This tool will run xdg autostart .desktop files

positional arguments:
  ENVIRONMENT  Optionally specify a list of environments for which to run
               autostart. If none are specified, only applications which do
               not limit themselves to certain environments will be run. Valid
               choices are the words GNOME, KDE, ROX, XFCE, and Old. (default:
               [])

optional arguments:
  -h, --help   show this help message and exit
  -l, --list   Show a list of the files which would be run. Files which would
               be run are marked with an asterix symbol [*]. For files which
               would not be run, information is given for why they are
               excluded (default: False)
```

Also, in fact I'm not quite sure if the appended [] to the "choose from..." list is completely harmless:
```
john@lithium:~$ bl-xdg-autostart []
usage: bl-xdg-autostart [-h] [-l] [ENVIRONMENT [ENVIRONMENT ...]]
bl-xdg-autostart: error: argument ENVIRONMENT: invalid choice: '[]' (choose from 'GNOME', 'KDE', 'ROX', 'XFCE', 'Old', [])
```
Though, if there's no simple, elegant fix, let's let it go.

johnraff commented on bin/bl-xdg-autostart:
```diff
@@ -21,27 +20,39 @@
+def main() -> int:
+    ap = ArgumentParser(
+        description="This tool will run xdg autostart .desktop files",
+        formatter_class=ArgumentDefaultsHelpFormatter,
```
Suggest removing this line, as the argument help `(default: *)` items are confusing.

johnraff replied:
"This line" refers to line 34 in case GitHub does not make that clear.
Removing it eliminates the (default: *) items I referred to above.

johnraff replied:
Line 23 import of ArgumentDefaultsHelpFormatter would presumably also be unnecessary.

2ion replied:
I always use the DefaultsHelpFormatter but it indeed might be confusing if somebody doesn't know what the defaults: statements mean ("flag is of course false", "default environment list is of course an empty list"). You can drop it.

johnraff committed:
Remove (Default: *) items from help output.

Re discussion: https://github.com/BunsenLabs/bunsen-configs/pull/109#pullrequestreview-497132942

johnraff commented:
If there are no further issues, I'll merge this tomorrow.
